src/components/structures/MatrixChat.tsx

```diff
@@ -1268,6 +1273,14 @@ export default class MatrixChat extends React.PureComponent<IProps, IState> {
         }
     }
 
+    private onInitialSync = (): void => {
+        if (this.cli.isInitialSyncComplete()) {
```

Comment:
I'm not sure this'll do what you want it to do. This will return true if a cached sync has been loaded, when we're not yet synced up with the server. So could end up overwriting another account_data event due to a race condition.

Reply:
Hmm good point, I've moved this into `Notifier` and wait for the first non-cached sync.